pydra/mark/tests/test_shell.py

```diff
@@ -0,0 +1,467 @@
+            )
+            y: str = shell_arg(
+                help_string="path of output file",
+                output_file_template="{x}_out",
+                output_field_name="y_out",
+                argstr="",
+            )
+
+    assert "y_out" in [a.name for a in attrs.fields(A.Outputs)]
+
+
+def test_shell_output_field_name_dynamic():
+    A = shell_task(
+        "A",
+        executable="cp",
+        input_fields={
+            "x": {
+                "type": os.PathLike,
+                "help_string": "an input file",
+                "argstr": "",
+                "position": 0,
+            },
+            "y": {
+                "type": str,
+                "help_string": "path of output file",
+                "argstr": "",
+                "output_field_name": "y_out",
+                "output_file_template": "{x}_out",
+            },
+        },
+    )
+
+    assert "y_out" in [a.name for a in attrs.fields(A.Outputs)]
+
+
+def get_file_size(y: Path):
+    result = os.stat(y)
+    return result.st_size
+
+
+def test_shell_bases_dynamic(A, tmpdir):
+    B = shell_task(
+        "B",
+        output_fields={
+            "out_file_size": {
+                "type": int,
+                "help_string": "size of the output directory",
+                "callable": get_file_size,
+            }
+        },
+        bases=[A],
+    )
+
+    xpath = tmpdir / "x.txt"
+    ypath = tmpdir / "y.txt"
+    Path.touch(xpath)
+
+    b = B(x=xpath, y=str(ypath))
+
+    result = b()
+
+    assert b.inputs.x == xpath
+    assert result.output.y == str(ypath)
+
+
+def test_shell_bases_static(A, tmpdir):
+    @shell_task
+    class B(A):
+        class Outputs:
+            out_file_size: int = shell_out(
+                help_string="size of the output directory", callable=get_file_size
+            )
+
+    xpath = tmpdir / "x.txt"
+    ypath = tmpdir / "y.txt"
+    Path.touch(xpath)
+
+    b = B(x=xpath, y=str(ypath))
+
+    result = b()
+
+    assert b.inputs.x == xpath
+    assert result.output.y == str(ypath)
+
+
+def test_shell_inputs_outputs_bases_dynamic(tmpdir):
+    A = shell_task(
+        "A",
+        "ls",
+        input_fields={
+            "directory": {
+                "type": os.PathLike,
+                "help_string": "input directory",
+                "argstr": "",
+                "position": -1,
+            }
+        },
+        output_fields={
+            "entries": {
+                "type": list,
+                "help_string": "list of entries returned by ls command",
+                "callable": list_entries,
+            }
+        },
+    )
+    B = shell_task(
+        "B",
+        "ls",
+        input_fields={
+            "hidden": {
+                "type": bool,
+                "argstr": "-a",
+                "help_string": "show hidden files",
+                "default": False,
+            }
+        },
+        bases=[A],
+        inputs_bases=[A.Inputs],
```

Comment:
not sure if I understand the idea behind this syntax (i.e. separately providing `bases` and `inputs_bases`